### PR TITLE
nim: add page

### PR DESCRIPTION
**Pull request by Calinou**
- [x] The page (if new), does not already exist in the repo.

- [x] The page (if new), has been added to the correct platform folder:
      `common/` if it's common to all platforms, `linux/` if it's Linux-specific, and so on.

- [x] The page has 8 or fewer examples.

- [x] The PR is appropriately titled:
      `<command name>: add page` for new pages, or `<command name>: <description of changes>` for pages being edited.

- [x] The page follows the [contributing](https://github.com/tldr-pages/tldr/blob/master/CONTRIBUTING.md) guidelines.

This adds a page for the [Nim](https://nim-lang.org/) compiler.

### CODE REVIEW

File: pages/common/nim.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
+# nim
+
+> The Nim compiler.
+> Processes, compiles and links Nim language source files.
+
+- Compile a source file:
+
+`nim compile {{file}}`
```
Comment (sbrl) [inline pages/common/nim.md]: Maybe `{{path/to/file.ext}}`, where `.ext` is the default extension for a nim source file? Also on the other examples below.

File: pages/common/nim.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+- Compile a source file with release optimizations enabled:
+
+`nim compile -d:release {{file}}`
+
+- Build a release binary optimized for low file size:
+
+`nim compile -d:release --opt:size {{file}}`
+
+- Generate HTML documentation for a module:
```
Comment (sbrl) [inline pages/common/nim.md]: Where does it put the documentation?

File: pages/common/nim.md (hunk 1 above)
Comment (Calinou) [reply pages/common/nim.md]: The `.nim` extension can be ommitted, e.g. a file called `main.nim` can be compiled using `nim compile main`.

File: pages/common/nim.md (hunk 2 above)
Comment (Calinou) [reply pages/common/nim.md]: The documentation is placed in the same directory as the module source file. Should I note that in the example?

File: pages/common/nim.md (hunk 1 above)
Comment (agnivade) [reply pages/common/nim.md]: But can a file called main.txt be compiled using `nim compile main` ? If not, then I think we should add the extension.

Comment (Calinou) [reply pages/common/nim.md]: @agnivade No, it only detects the `.nim` extension; I think it makes sense to add the extension in the example.

**Comment by agnivade** [comment]
We actually prefer spelled out options rather than short ones. For eg. we use `npm install`, rather than `npm i`. Once the user has grasped the command, he/she is free to use shorter options.

**Pull request merged**